### ISSUE TITLE
Unexpected ref safety error for a prefix user-defined increment operator

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#78964]
```
public ref struct C
{
    public C M1(C c, scoped C c1)
    {
        c = c1++; // error CS8352: Cannot use variable 'scoped C c1' in this context because it may expose referenced variables outside of their declaration scope
        return c;
    }

    public C M2(C c, scoped C c2)
    {
        c = ++c2; // Unexpected error CS8352: Cannot use variable 'scoped C c2' in this context because it may expose referenced variables outside of their declaration scope
        return c;
    }
    
    public static C operator ++(scoped C left) => throw null;
}
```

**Comment by jjonescz** [comment]
EDIT: Nevermind, the lowering of `c = ++c2;` is probably more equivalent to `var tmp = C.Inc(c2); c2 = tmp; c = tmp` and so it really shouldn't result in a ref safety error.

Original comment:

This looks expected to me. Consider the equivalent method calls would result in the same errors:
```cs
public ref struct C
{
    public C M1(C c, scoped C c1)
    {
        c = c1++;
        // equivalent to:
        var tmp = c1;
        c1 = C.Inc(c1);
        c = tmp;
        return c;
    }

    public C M2(C c, scoped C c2)
    {
        c = ++c2;
        // equivalent to:
        c = (c2 = C.Inc(c2));
        return c;
    }
    
    public static C operator ++(scoped C left) => throw null;
    public static C Inc(scoped C left) => throw null;
}
```

In particular, the `error CS8352: Cannot use variable 'scoped C c2' in this context because it may expose referenced variables outside of their declaration scope` is expected because `c2` (which is scoped) cannot be assigned to `c` (which is unscoped).
